api/islandora_xacml_api.drush.inc

```diff
@@ -135,3 +157,144 @@ EOQ;
+  drupal_static_reset('islandora_get_tuque_connection');
+  $tuque = islandora_get_tuque_connection();
+
+  $query = <<<EOQ
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+SELECT DISTINCT ?object ?created FROM <#ri>
+WHERE {
+  ?object <fedora-view:disseminates> ?ds ;
+          <fedora-view:disseminates> ?ds2 ;
+          <fedora-model:createdDate> ?created .
+  ?ds <fedora-view:disseminationType> <info:fedora/*/POLICY> .
+  {
+    ?ds2 <http://islandora.ca/ontology/relsint#isViewableByRole> "administrator"
+  }
+  UNION
+  {
+    ?ds2 <http://islandora.ca/ontology/relsint#isViewableByUser> "fedoraAdmin"
+  }
+  !filter
+}
+ORDER BY ?created
+
+EOQ;
+  if (!isset($sandbox['total'])) {
+    $sandbox['progress'] = 0;
+    $sandbox['total'] = $tuque->repository->ri->countQuery(format_string($query, array(
+      '!filter' => '',
+    )), 'sparql');
+    if ($sandbox['total'] == 0) {
+      $context['finished'] = 1;
+      $context['message'] = t('Nothing to verify.');
+      return;
+    }
+    $sandbox['result_stash'] = array();
+  }
+  if (empty($sandbox['result_stash'])) {
+    if (isset($sandbox['last_date'])) {
+      $filter = format_string('FILTER(?created > "!created"^^xsd:dateTime || (?created = "!created"^^xsd:dateTime && xsd:string(?object) > "info:fedora/!pid"^^xsd:string))', array(
+        '!created' => $sandbox['last_date'],
+        '!pid' => $sandbox['last_pid'],
+      ));
+    }
+    $limited_query = format_string($query, array(
+      '!filter' => isset($filter) ? $filter : '',
+    )) . <<<EOQ
+LIMIT $limit
+EOQ;
+
+    drush_log(dt('Querying for @count objects.', array('@count' => $limit)));
+    $sandbox['result_stash'] = $tuque->repository->ri->sparqlQuery($limited_query);
+    if (empty($sandbox['result_stash'])) {
+      $context['finished'] = 1;
+      $context['message'] = t('Updated @count PID(s) and ran out of items early. POLICY was manually removed?', array(
+        '@count' => $sandbox['progress'],
+      ));
+      return;
+    }
+  }
+
+  $result = array_shift($sandbox['result_stash']);
+  $sandbox['last_date'] = $result['created']['value'];
+  $sandbox['last_pid'] = $result['object']['value'];
+  $object = islandora_object_load($result['object']['value']);
+  drush_log(dt('Processing @pid.', array(
+    '@pid' => $object->id,
+  )));
+  // Get the POLICY and compare it to the RELS-INT statements.
+  $xacml = new IslandoraXacml($object);
+  $to_update = FALSE;
+  $datastream_data = $xacml->datastreamRule->getRuleArray();
+  foreach ($datastream_data['dsids'] as $dsid) {
+    // Only check DSIDs that exist on the object given the nature of how the API
+    // writes RELS-INT statements when that datastream gets ingested.
+    if (isset($object[$dsid])) {
+      // Ensure the users and roles match up.
+      foreach ($datastream_data['users'] as $user) {
+        $user_rels = $object[$dsid]->relationships->get(ISLANDORA_RELS_INT_URI, 'isViewableByUser', $user, RELS_TYPE_PLAIN_LITERAL);
```

Comment:
only isViewableByRole and isViewableByUser was affected? any other predicate?

Reply:
These are just identifying conditions based on the defaults set in https://github.com/Islandora/islandora_xacml_editor/blob/7.x/api/includes/xacml.inc#L277-L285. RELS-INT has no isManageableBy predicate as per https://jira.duraspace.org/browse/ISLANDORA-1273.

Reply:
Ok, that answers my question. cool